### ISSUE TITLE
[firefox] "[fresh] [error] n2p_call_ptac, NPN_Invoke failed (or there were no npp)"

### ISSUE THREAD

**Issue by BlackMage2** [i-rinat/freshplayerplugin#369]
the FlashPlayer is used e.g. not recognized in DarkOrbit
it comes messages "[fresh] [error] n2p_call_ptac, NPN_Invoke failed (or there were no npp)" in console and in the browser comes the error message that you need the latest Flash Player to play

**Comment by i-rinat** [comment]
I tried to open that game. It complains about Flash player being old, but work. Although, slow, very slow, and with enormous CPU usage, as it seems to use CPU rendering. It doesn't complain in Chrome, which use the same PepperFlash version. So it could be missing features.

As for graphics, I currently have two features unimplemented: `PPB_OpenGLES2VertexArrayObject;1.0` and `PPB_OpenGLES2DrawBuffers(Dev);1.0`. I'm not really sure whenever they are the key.

**Comment by BlackMage2** [comment]
sure?
can you also start the game?

**Comment by i-rinat** [comment]
> how can you start the game?

Yes, it works. At least I think it works, as I can move the ship in various directions. But it also looks like I'm locked to `ru3.darkorbit.com` server. If I try to open `int1.darkorbit.com`, it still redirects to ru3. It's also possible that various servers use slightly different versions. So the game working for me doesn't solve anything.

Anyway, it's still unplayable, as there is almost a one second lag after almost every mouse click. I'd be enraged after a dozen of minutes. It's definitely not hardware-related, as gameplay is very smooth in Chrome. So, if you want to play, I'd suggest using Chrome (or Chromium), as I doubt that the issue is easily fixable in freshplayerplugin.

**Comment by i-rinat** [comment]
> appears with the plugin version "0.0 r0"

Try to remove `pluginreg.dat` file. It's in the Firefox profile directory.
Firefox caches plugin information in that file, to speed-up startup. Sometimes for some reason some of the plugins are marked as faulty. From that point, Firefox won't try to load plugin again. Removing `pluginreg.dat` will force Firefox to scan plugins again. You'll need to restart the browser.

**Comment by BlackMage2** [comment]
pluginreg.dat deleted and no change

**Comment by i-rinat** [comment]
> the plugin version is still displayed as "0.0 r0"

Is there `manifest.json` file near `libpepflashplayer.so`? Manifest file is used to determine version of Flash. If there is no such file for some reason, version would be all-zeros.

> does not recognize the Flash Player so I can't play

In Chrome, before, there were possibility to make it ask whenever you want Flash on a particular page. Now there are only two options. First blocks Flash completely. Second allows Flash, but only on sites you explicitly specify in settings. (Open chrome://settings/, scroll down to Advanced, open it, scroll down to Content Settings, find there Flash, add site addresses to the Allowed list).
